include/vinecopulib/bicop/class.hpp

```diff
@@ -119,24 +115,40 @@ class Bicop
 
   Eigen::MatrixXd get_parameters_upper_bounds() const;
 
+  Bicop as_continuous() const;
+
 private:
-  Eigen::Matrix<double, Eigen::Dynamic, 2> cut_and_rotate(
-    const Eigen::Matrix<double, Eigen::Dynamic, 2>& u) const;
+  Eigen::MatrixXd format_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd clip_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd rotate_data(const Eigen::MatrixXd& u) const;
+
+  Eigen::MatrixXd prep_for_abstract(const Eigen::MatrixXd& u) const;
 
   void check_rotation(int rotation) const;
 
+  void check_data(const Eigen::MatrixXd& u) const;
+
+  void check_data_dim(const Eigen::MatrixXd& u) const;
+
+  void flip_var_types();
+
   void check_weights_size(const Eigen::VectorXd& weights,
                           const Eigen::MatrixXd& data) const;
 
   void check_fitted() const;
 
+  int get_n_discrete() const;
```

Comment:
Following the trace of where this function was used made me think: currently, we use `format_data` everytime we try to recompute the log-likelihood in the fit. It's really super inefficient, when has this changed? At some point, we had a way to compute a "rough" log-likelihood that was not re-formatting the data at every function call...